### PR TITLE
[Swift] basics - replace about.md file with concept files

### PR DESCRIPTION
**Pull request by ErikSchierboom**
In [this issue](https://github.com/exercism/v3/issues/2293) we're describing an evolution of the specification of this repo.
These changes include, amongst others, replacing the contents of the `after.md` document with individual concept documents. There are two documents per concept:

1. An `about.md` file containing the description of the concept
1. A `links.json` file containing a set of useful links related to the concept. 

This looks as follows in the repo:

```
<track>
├── concepts
│   ├── <concept>
│   │   ├── about.md
│   │   └── links.json
│   └── ...
```

This PR applies this change to this exercise by: 
                    
1. Copying the existing `after.md` file's contents to the `about.md` file of each concept listed in the exercise's `concepts` section in the `config.json` file.
1. Extract the links from the `after.md` file and put them into the `links.json` file

Before merging this PR, please check that the contents of the concept documents makes sense.
This is especially true when the exercise unlocks multiple concepts, as then the concept documents should be updated to only contain the information relevant to that concept.

Note: to make reviewing easier, the PR has been split into three separate commits. Please squash this PR upon merging.

### CODE REVIEW

**Comment by wneumann** [comment]
Wow. This is… not good. It's really just making me regret having put in all the work I had before this point.

Oh well. I guess it's time to roll up the sleeves and see what's needed to fix this mess.

**Comment by ErikSchierboom** [comment]
Yeah sorry about this @wneumann!

**Comment by wneumann** [comment]
OK, so… In theory I have cleaned up the individual about.md files to contain just their relevant bits.
I deleted the `variables` concept and made a combined `constants-and-variables` concept as it seems to make more sense to talk about them together.
I also blanked out all but one of the links files as those are just the links in the text and not additional potentially-helpful links (which is what I get from https://github.com/exercism/v3/issues/2499). Those can be added later, perhaps in separate PRs.

@ErikSchierboom, take a look and let me know if this is in the spirit of this reorg, and I'll continue with the other exercises.

**Review by ErikSchierboom — COMMENTED** [review]
@wneumann So in this case what you could also do is to, instead of having several concepts here, just have one `basics` exercise. The `basics` exercise is a bit "special" in that sense, for other exercises one definitely wants to split these up. That said, I'm totally fine with having them like this for now and seeing how it pans out.

CC @iHiD

File: languages/swift/concepts/functions/about.md [hunk 1]
```diff
@@ -1 +1,52 @@
-TODO: add information on functions concept
+In Swift, [functions][functions] are also values that are associated with names, though the syntax used to declare a function differs from that used to declare a constant or a variable. They are defined using the `func` keyword followed by a pair of parentheses enclosing a comma-separated list of parameter names where these names include an [argument label and a parameter name][argument labels] followed by a [type annotation][type annotations]. The argument label is used to refer to a parameter when the function is _called_ and the parameter name is used to refer to the parameter's value inside the function body. The parameter list is followed by `->` and the type of the value returned by the function. Note that the type annotations for the parameters are required and can not be inferred by the compiler. If the return type of the function is `Void`, i.e. `()`, the `-> ()` or `-> Void` may be omitted.
```
Comment (ErikSchierboom) [inline languages/swift/concepts/functions/about.md:1]: It looks like the `[type annotations]` link is not defined.

File: languages/swift/config.json [hunk 2]
```diff
@@ -21,10 +21,9 @@
         "uuid": "44ae5892-40a5-4483-a181-28e1c38645d6",
         "concepts": [
           "basics",
-          "constants",
+          "constants-and-variables",
```
Comment (ErikSchierboom) [inline languages/swift/config.json:24]: I like the rename!

File: languages/swift/concepts/functions/about.md (hunk 1 above)
Comment (wneumann) [reply languages/swift/concepts/functions/about.md:1]: I added it back in in the last commit

**Comment by ErikSchierboom** [comment]
@wneumann It looks like I screwed up this PR :(

**Comment by wneumann** [comment]
@ErikSchierboom As noted in the lasagna-master PR, I think it's easiest to just close this PR and replace it with https://github.com/exercism/v3/pull/2668 which replaces the after.md file with the concept files, but does it from the renamed exercise.

If you concur, we can close this PR and go with 2668.

**Review by SleeplessByte — CHANGES_REQUESTED** [review]
I think you accidentally comited those files you made for me (`js/concepts/**/links.json`)

**Pull request closed without merging**